src/psyclone/tests/psyGen_test.py

```diff
@@ -3173,6 +3173,64 @@ def test_reference_can_be_printed():
     assert "Reference[name:'rname']" in str(ref)
 
 
+def test_reference_symbol(monkeypatch):
+    '''Test that the symbol method in a Reference Node instance returns
+    the associated symbol if there is one and None if not. Also test
+    for an incorrect scope argument.
+
+    '''
+    _, invoke = get_invoke("single_invoke_kern_with_global.f90",
+                           api="gocean1.0", idx=0)
+    sched = invoke.schedule
+    kernels = sched.walk(Kern)
+    kernel_schedule = kernels[0].get_kernel_schedule()
+    references = kernel_schedule.walk(Reference)
+
+    # Symbol in KernelSchedule SymbolTable
+    field_old = references[0]
+    assert field_old.name == "field_old"
+    assert isinstance(field_old.symbol(), Symbol)
+    assert field_old.symbol().name == field_old.name
+
+    # Symbol in KernelSchedule SymbolTable with KernelSchedule scope
+    assert isinstance(field_old.symbol(scope_limit=kernel_schedule), Symbol)
+    assert field_old.symbol().name == field_old.name
+
+    # Symbol in KernelSchedule SymbolTable with parent scope
+    assert field_old.symbol(scope_limit=field_old.parent) is None
+
+    # Symbol in Container SymbolTable
+    alpha = references[6]
+    assert alpha.name == "alpha"
+    assert isinstance(alpha.symbol(), Symbol)
+    assert alpha.symbol().name == alpha.name
+
+    # Symbol in Container SymbolTable with KernelSchedule scope
+    assert alpha.symbol(scope_limit=kernel_schedule) is None
+
+    # Symbol in Container SymbolTable with Container scope
+    assert isinstance(kernel_schedule.root, Container)
+    assert alpha.symbol(scope_limit=kernel_schedule.root).name == alpha.name
+
+    # Symbol method with invalid scope type
+    with pytest.raises(GenerationError) as excinfo:
```

Comment:
It feels that this one should be a TypeError (but the test below is rightly a GenerationError) I am happy to overlook this issue if we consider the function as internal and the type checking is not needed.

Reply:
Updated.